pkg/driver/controller.go

```diff
@@ -154,9 +154,14 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 		}
 	}
 
+	snapshotID := ""
+	if snapshotSource := req.VolumeContentSource.GetSnapshot(); snapshotSource != nil {
+		snapshotID = snapshotSource.SnapshotId
```

Comment:
Do we need to set the snapshot ID for an existing volume? Or only at line 188 here 
https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/446/files#diff-61566437d9b17d18936960aa0bce1ad4R187-R196

for restoring/creating the volume out of snapshot

Reply:
Yes, for idempotency. Responses to repeated CreateVolume requests for snapshot restore must have the same fields as the initial response.